src/audio/drc/drc.c

```diff
@@ -140,353 +141,206 @@ static int drc_setup(struct drc_comp_data *cd, uint16_t channels, uint32_t rate)
-		break;
-	default:
-		comp_err(dev, "drc_cmd(), invalid command");
-		ret = -EINVAL;
-	}
-
-	return ret;
-}
-
-static int drc_trigger(struct comp_dev *dev, int cmd)
-{
-	int ret;
+	struct sof_ipc_ctrl_data *cdata = (struct sof_ipc_ctrl_data *)fragment;
+	struct drc_comp_data *cd = module_get_private_data(mod);
 
-	comp_info(dev, "drc_trigger()");
+	comp_info(mod->dev, "drc_get_config()");
 
-	ret = comp_set_state(dev, cmd);
-	if (ret == COMP_STATUS_STATE_ALREADY_SET)
-		ret = PPL_STATUS_PATH_STOP;
-
-	return ret;
+	return comp_data_blob_get_cmd(cd->model_handler, cdata, fragment_size);
 }
 
-static void drc_process(struct comp_dev *dev, struct comp_buffer __sparse_cache *source,
-			struct comp_buffer __sparse_cache *sink, int frames,
-			uint32_t source_bytes, uint32_t sink_bytes)
+static void drc_set_alignment(struct audio_stream __sparse_cache *source,
+			      struct audio_stream __sparse_cache *sink)
 {
-	struct drc_comp_data *cd = comp_get_drvdata(dev);
-
-	buffer_stream_invalidate(source, source_bytes);
-
-	cd->drc_func(dev, &source->stream, &sink->stream, frames);
-
-	buffer_stream_writeback(sink, sink_bytes);
-
-	/* calc new free and available */
-	comp_update_buffer_consume(source, source_bytes);
-	comp_update_buffer_produce(sink, sink_bytes);
+	/* Currently no optimizations those would use wider loads and stores */
+	audio_stream_init_alignment_constants(1, 1, source);
+	audio_stream_init_alignment_constants(1, 1, sink);
 }
 
-/* copy and process stream data from source to sink buffers */
-static int drc_copy(struct comp_dev *dev)
+static int drc_process(struct processing_module *mod,
+		       struct input_stream_buffer *input_buffers,
+		       int num_input_buffers,
+		       struct output_stream_buffer *output_buffers,
+		       int num_output_buffers)
 {
-	struct comp_copy_limits cl;
-	struct drc_comp_data *cd = comp_get_drvdata(dev);
-	struct comp_buffer *sourceb, *sinkb;
-	struct comp_buffer __sparse_cache *source_c, *sink_c;
-	int ret = 0;
-
-	comp_dbg(dev, "drc_copy()");
-
-	sourceb = list_first_item(&dev->bsource_list, struct comp_buffer,
-				  sink_list);
-	sinkb = list_first_item(&dev->bsink_list, struct comp_buffer,
-				source_list);
+	struct drc_comp_data *cd = module_get_private_data(mod);
+	struct comp_dev *dev = mod->dev;
+	struct audio_stream __sparse_cache *source = input_buffers[0].data;
+	struct audio_stream __sparse_cache *sink = output_buffers[0].data;
+	int frames = input_buffers[0].size;
+	int ret;
 
-	source_c = buffer_acquire(sourceb);
-	sink_c = buffer_acquire(sinkb);
+	comp_dbg(dev, "drc_process()");
 
 	/* Check for changed configuration */
 	if (comp_is_new_data_blob_available(cd->model_handler)) {
 		cd->config = comp_get_data_blob(cd->model_handler, NULL, NULL);
-		ret = drc_setup(cd, audio_stream_get_channels(&source_c->stream),
-				audio_stream_get_rate(&source_c->stream));
+		ret = drc_setup(cd, audio_stream_get_channels(sink), audio_stream_get_rate(sink));
```

Comment:
you flip from source to sink here, why?

Reply:
I think the flip should be contained by the following commit https://github.com/thesofproject/sof/pull/7621/commits/90997e3af60d4b29a8dfe224c74ccae8636610a8 instead

Reply:
Fixed now.